### DISCUSSION TITLE
Implicit perfection/imperfection of rhythmic values in mensural music.

### DISCUSSION THREAD

**Discussion by craigsapp** [rism-digital/verovio#3519] [General]
In this example:

<img width="506" alt="Screenshot 2023-09-25 at 06 55 01" src="https://github.com/rism-digital/verovio/assets/3487289/c6d142d2-1d90-4385-96c9-4f05cc47aede">

<details>
<summary> Click to view MEI data for above example </summary>

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/5.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/5.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="5.0">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title />
   </titleStmt>
   <pubStmt />
  </fileDesc>
  <encodingDesc>
   <appInfo>
    <application isodate="2023-09-25T06:54:46" version="4.1.0-dev-7518ccb-dirty">
     <name>Verovio</name>
     <p>Transcoded from Humdrum</p>
    </application>
   </appInfo>
  </encodingDesc>
  <workList>
   <work>
    <title />
   </work>
  </workList>
 </meiHead>
 <music>
  <body>
   <mdiv xml:id="mj3u2ql">
    <score xml:id="sad5u2a">
     <scoreDef xml:id="s1bs5ghf" midi.bpm="400.000000">
      <staffGrp xml:id="s16qo4su" bar.thru="true" symbol="brace">
       <staffDef xml:id="staffdef-L1F2" n="1" notationtype="mensural.white" lines="5">
        <label xml:id="label-L5F2">Canto</label>
        <labelAbbr xml:id="labelabbr-L6F2">C</labelAbbr>
        <clef xml:id="clef-L7F2" shape="C" line="1" />
        <keySig xml:id="keysig-L9F2" sig="0" />
        <instrDef xml:id="ivs2ma6" midi.instrnum="52" midi.instrname="Choir_Aahs" />
       </staffDef>
       <staffDef xml:id="staffdef-L1F1" n="2" notationtype="mensural.white" lines="5">
        <label xml:id="label-L5F1">Alto</label>
        <labelAbbr xml:id="labelabbr-L6F1">A</labelAbbr>
        <clef xml:id="clef-L7F1" shape="C" line="3" />
        <keySig xml:id="keysig-L9F1" sig="0" />
        <instrDef xml:id="i11h4nqo" midi.instrnum="52" midi.instrname="Choir_Aahs" />
       </staffDef>
      </staffGrp>
     </scoreDef>
     <section xml:id="section-L1F1">
      <staff xml:id="staff-L1F2" n="1">
       <layer xml:id="layer-L1F2N1" n="1">
        <mensur xml:id="mensur-L10F2" sign="C" slash="1" />
        <rest xml:id="rest-L13F2" dur="minima" />
        <note xml:id="note-L14F2" dur="minima" oct="4" pname="a" />
        <note xml:id="note-L15F2" dur="minima" oct="4" pname="d" />
        <note xml:id="note-L16F2" dur="minima" oct="4" pname="d" />
        <note xml:id="note-L17F2" dur="minima" oct="4" pname="e" />
        <note xml:id="note-L18F2" dur="minima" oct="4" pname="f">
         <accid xml:id="accid-L18F2" accid="s" />
        </note>
        <note xml:id="note-L19F2" dur="minima" oct="4" pname="g" />
        <note xml:id="note-L20F2" dur="minima" oct="4" pname="e" />
        <barLine xml:id="barline-L21F2" form="dbl" />
       </layer>
      </staff>
      <staff xml:id="staff-L1F1" n="2">
       <layer xml:id="layer-L1F1N1" n="1">
        <mensur xml:id="mensur-L10F1" sign="C" slash="1" />
        <rest xml:id="rest-L13F1" dur="semibrevis" />
        <rest xml:id="rest-L15F1" dur="minima" />
        <note xml:id="note-L16F1" dur="minima" oct="4" pname="d" />
        <note xml:id="note-L17F1" dur="minima" oct="3" pname="a" />
        <note xml:id="note-L18F1" dur="minima" oct="3" pname="a" />
        <note xml:id="note-L19F1" dur="minima" oct="3" pname="b" />
        <note xml:id="note-L20F1" dur="minima" oct="4" pname="c">
         <accid xml:id="accid-L20F1" accid="s" />
        </note>
        <barLine xml:id="barline-L21F1" form="dbl" />
       </layer>
      </staff>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```

</details>

The semibreve (whole note) rest at the start of the bottom staff is being implicitly assigned `@dur.quality="perfecta"` when it should be `@dur.quality="imperfecta"` due to the Cut-C (or C) mensuration sign.   Adding `@dur.quality="imperfecta"` to the semibreve rest causes its duration to be correct:

<img width="525" alt="Screenshot 2023-09-25 at 07 24 41" src="https://github.com/rism-digital/verovio/assets/3487289/a71458da-be62-4d89-9601-74b20fc6fbec">

<details>
<summary> Click to view MEI data for above example </summary>

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/5.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/5.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="5.0">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title />
   </titleStmt>
   <pubStmt />
  </fileDesc>
  <encodingDesc>
   <appInfo>
    <application isodate="2023-09-25T06:54:46" version="4.1.0-dev-7518ccb-dirty">
     <name>Verovio</name>
     <p>Transcoded from Humdrum</p>
    </application>
   </appInfo>
  </encodingDesc>
  <workList>
   <work>
    <title />
   </work>
  </workList>
 </meiHead>
 <music>
  <body>
   <mdiv xml:id="mj3u2ql">
    <score xml:id="sad5u2a">
     <scoreDef xml:id="s1bs5ghf" midi.bpm="400.000000">
      <staffGrp xml:id="s16qo4su" bar.thru="true" symbol="brace">
       <staffDef xml:id="staffdef-L1F2" n="1" notationtype="mensural.white" lines="5">
        <label xml:id="label-L5F2">Canto</label>
        <labelAbbr xml:id="labelabbr-L6F2">C</labelAbbr>
        <clef xml:id="clef-L7F2" shape="C" line="1" />
        <keySig xml:id="keysig-L9F2" sig="0" />
        <instrDef xml:id="ivs2ma6" midi.instrnum="52" midi.instrname="Choir_Aahs" />
       </staffDef>
       <staffDef xml:id="staffdef-L1F1" n="2" notationtype="mensural.white" lines="5">
        <label xml:id="label-L5F1">Alto</label>
        <labelAbbr xml:id="labelabbr-L6F1">A</labelAbbr>
        <clef xml:id="clef-L7F1" shape="C" line="3" />
        <keySig xml:id="keysig-L9F1" sig="0" />
        <instrDef xml:id="i11h4nqo" midi.instrnum="52" midi.instrname="Choir_Aahs" />
       </staffDef>
      </staffGrp>
     </scoreDef>
     <section xml:id="section-L1F1">
      <staff xml:id="staff-L1F2" n="1">
       <layer xml:id="layer-L1F2N1" n="1">
        <mensur xml:id="mensur-L10F2" sign="C" slash="1" />
        <rest xml:id="rest-L13F2" dur="minima" />
        <note xml:id="note-L14F2" dur="minima" oct="4" pname="a" />
        <note xml:id="note-L15F2" dur="minima" oct="4" pname="d" />
        <note xml:id="note-L16F2" dur="minima" oct="4" pname="d" />
        <note xml:id="note-L17F2" dur="minima" oct="4" pname="e" />
        <note xml:id="note-L18F2" dur="minima" oct="4" pname="f">
         <accid xml:id="accid-L18F2" accid="s" />
        </note>
        <note xml:id="note-L19F2" dur="minima" oct="4" pname="g" />
        <note xml:id="note-L20F2" dur="minima" oct="4" pname="e" />
        <barLine xml:id="barline-L21F2" form="dbl" />
       </layer>
      </staff>
      <staff xml:id="staff-L1F1" n="2">
       <layer xml:id="layer-L1F1N1" n="1">
        <mensur xml:id="mensur-L10F1" sign="C" slash="1" />
        <rest xml:id="rest-L13F1" dur="semibrevis" dur.quality="imperfecta" />
        <rest xml:id="rest-L15F1" dur="minima" />
        <note xml:id="note-L16F1" dur="minima" oct="4" pname="d" />
        <note xml:id="note-L17F1" dur="minima" oct="3" pname="a" />
        <note xml:id="note-L18F1" dur="minima" oct="3" pname="a" />
        <note xml:id="note-L19F1" dur="minima" oct="3" pname="b" />
        <note xml:id="note-L20F1" dur="minima" oct="4" pname="c">
         <accid xml:id="accid-L20F1" accid="s" />
        </note>
        <barLine xml:id="barline-L21F1" form="dbl" />
       </layer>
      </staff>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```

</details>

However, in the following test, all notes are being treated correctly as imperfecta:

<img width="953" alt="Screenshot 2023-09-25 at 07 35 10" src="https://github.com/rism-digital/verovio/assets/3487289/756ab390-39f9-4cda-a0a1-15b766f28bde">

<details>
<summary> Click to view MEI data for above example </summary>

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/5.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/5.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="5.0">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title />
   </titleStmt>
   <pubStmt />
  </fileDesc>
  <encodingDesc>
   <appInfo>
    <application isodate="2023-09-25T07:34:23" version="4.1.0-dev-7518ccb-dirty">
     <name>Verovio</name>
     <p>Transcoded from Humdrum</p>
    </application>
   </appInfo>
  </encodingDesc>
  <workList>
   <work>
    <title xml:id="title-L38" analog="humdrum:Xfi" type="translated">kern2mens</title>
   </work>
  </workList>
 </meiHead>
 <music>
  <body>
   <mdiv xml:id="m1plzmwq">
    <score xml:id="s1hizj">
     <scoreDef xml:id="s1dixjw7" midi.bpm="400.000000">
      <staffGrp xml:id="s130a6nl" bar.thru="true" symbol="brace">
       <staffDef xml:id="staffdef-L1F2" n="1" notationtype="mensural.white" lines="5">
        <clef xml:id="clef-L2F2" shape="C" line="3" />
       </staffDef>
       <staffDef xml:id="staffdef-L1F1" n="2" notationtype="mensural.white" lines="5">
        <clef xml:id="clef-L2F1" shape="C" line="3" />
       </staffDef>
      </staffGrp>
     </scoreDef>
     <section xml:id="section-L1F1">
      <staff xml:id="staff-L1F2" n="1">
       <layer xml:id="layer-L1F2N1" n="1">
        <mensur xml:id="mensur-L3F2" prolatio="2" tempus="2" sign="C" slash="1" />
        <note xml:id="note-L4F2" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L5F2" dur="minima" oct="4" pname="c" />
        <note xml:id="note-L7F2" dur="semibrevis" oct="4" pname="c" />
        <note xml:id="note-L11F2" dur="brevis" oct="4" pname="c" />
        <note xml:id="note-L19F2" dur="semibrevis" oct="4" pname="c" />
        <note xml:id="note-L23F2" dur="minima" oct="4" pname="c" />
        <note xml:id="note-L25F2" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L27F2" dur="minima" oct="4" pname="c" />
        <note xml:id="note-L28F2" dur="semibrevis" oct="4" pname="c" />
        <note xml:id="note-L30F2" dur="brevis" oct="4" pname="c" />
        <note xml:id="note-L34F2" dur="semibrevis" oct="4" pname="c" />
        <note xml:id="note-L36F2" dur="minima" oct="4" pname="c" />
       </layer>
      </staff>
      <staff xml:id="staff-L1F1" n="2">
       <layer xml:id="layer-L1F1N1" n="1">
        <mensur xml:id="mensur-L3F1" prolatio="2" tempus="2" sign="C" slash="1" />
        <note xml:id="note-L4F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L5F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L6F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L7F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L8F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L9F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L10F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L11F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L12F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L13F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L14F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L15F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L16F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L17F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L18F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L19F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L20F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L21F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L22F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L23F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L24F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L25F1" dur="semiminima" oct="4" pname="c" />
        <note xml:id="note-L27F1" dur="minima" oct="4" pname="c" />
        <note xml:id="note-L28F1" dur="minima" oct="4" pname="c" />
        <note xml:id="note-L29F1" dur="minima" oct="4" pname="c" />
        <note xml:id="note-L30F1" dur="minima" oct="4" pname="c" />
        <note xml:id="note-L31F1" dur="minima" oct="4" pname="c" />
        <note xml:id="note-L32F1" dur="minima" oct="4" pname="c" />
        <note xml:id="note-L33F1" dur="minima" oct="4" pname="c" />
        <note xml:id="note-L34F1" dur="minima" oct="4" pname="c" />
        <note xml:id="note-L35F1" dur="minima" oct="4" pname="c" />
        <note xml:id="note-L36F1" dur="minima" oct="4" pname="c" />
       </layer>
      </staff>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```

</details>

So I am confused as to why in the first example the semibreve rest is being treated as `perfecta`. (I tried converting to a note and it was still `perfecta`).

-----------------------------------------------

Preferably the Cut-C mensuration sign would implicitly set the tempus and prolatio level divisions to imperfect.
This can be explicitly set in the `<mensur>` by adding `@prolatio="2"` for the semibreve->minim level divisions, and `@tempus="2"` for the breve->semibreve level divisions:

<img width="532" alt="Screenshot 2023-09-25 at 07 21 14" src="https://github.com/rism-digital/verovio/assets/3487289/31ae2e4d-d2fb-4928-99cb-f95534e9ddaa">

<details>
<summary> Click to view MEI data for above example </summary>

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/5.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/5.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="5.0">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title />
   </titleStmt>
   <pubStmt />
  </fileDesc>
  <encodingDesc>
   <appInfo>
    <application isodate="2023-09-25T06:54:46" version="4.1.0-dev-7518ccb-dirty">
     <name>Verovio</name>
     <p>Transcoded from Humdrum</p>
    </application>
   </appInfo>
  </encodingDesc>
  <workList>
   <work>
    <title />
   </work>
  </workList>
 </meiHead>
 <music>
  <body>
   <mdiv xml:id="mj3u2ql">
    <score xml:id="sad5u2a">
     <scoreDef xml:id="s1bs5ghf" midi.bpm="400.000000">
      <staffGrp xml:id="s16qo4su" bar.thru="true" symbol="brace">
       <staffDef xml:id="staffdef-L1F2" n="1" notationtype="mensural.white" lines="5">
        <label xml:id="label-L5F2">Canto</label>
        <labelAbbr xml:id="labelabbr-L6F2">C</labelAbbr>
        <clef xml:id="clef-L7F2" shape="C" line="1" />
        <keySig xml:id="keysig-L9F2" sig="0" />
        <instrDef xml:id="ivs2ma6" midi.instrnum="52" midi.instrname="Choir_Aahs" />
       </staffDef>
       <staffDef xml:id="staffdef-L1F1" n="2" notationtype="mensural.white" lines="5">
        <label xml:id="label-L5F1">Alto</label>
        <labelAbbr xml:id="labelabbr-L6F1">A</labelAbbr>
        <clef xml:id="clef-L7F1" shape="C" line="3" />
        <keySig xml:id="keysig-L9F1" sig="0" />
        <instrDef xml:id="i11h4nqo" midi.instrnum="52" midi.instrname="Choir_Aahs" />
       </staffDef>
      </staffGrp>
     </scoreDef>
     <section xml:id="section-L1F1">
      <staff xml:id="staff-L1F2" n="1">
       <layer xml:id="layer-L1F2N1" n="1">
        <mensur xml:id="mensur-L10F2" sign="C" slash="1" tempus="2" prolatio="2" />
        <rest xml:id="rest-L13F2" dur="minima" />
        <note xml:id="note-L14F2" dur="minima" oct="4" pname="a" />
        <note xml:id="note-L15F2" dur="minima" oct="4" pname="d" />
        <note xml:id="note-L16F2" dur="minima" oct="4" pname="d" />
        <note xml:id="note-L17F2" dur="minima" oct="4" pname="e" />
        <note xml:id="note-L18F2" dur="minima" oct="4" pname="f">
         <accid xml:id="accid-L18F2" accid="s" />
        </note>
        <note xml:id="note-L19F2" dur="minima" oct="4" pname="g" />
        <note xml:id="note-L20F2" dur="minima" oct="4" pname="e" />
        <barLine xml:id="barline-L21F2" form="dbl" />
       </layer>
      </staff>
      <staff xml:id="staff-L1F1" n="2">
       <layer xml:id="layer-L1F1N1" n="1">
        <mensur xml:id="mensur-L10F1" sign="C" slash="1" tempus="2" prolatio="2" />
        <rest xml:id="rest-L13F1" dur="semibrevis" />
        <rest xml:id="rest-L15F1" dur="minima" />
        <note xml:id="note-L16F1" dur="minima" oct="4" pname="d" />
        <note xml:id="note-L17F1" dur="minima" oct="3" pname="a" />
        <note xml:id="note-L18F1" dur="minima" oct="3" pname="a" />
        <note xml:id="note-L19F1" dur="minima" oct="3" pname="b" />
        <note xml:id="note-L20F1" dur="minima" oct="4" pname="c">
         <accid xml:id="accid-L20F1" accid="s" />
        </note>
        <barLine xml:id="barline-L21F1" form="dbl" />
       </layer>
      </staff>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```

</details>

But preferably

```xml
        <mensur xml:id="mensur-L10F1" sign="C" slash="1" />
```

would be treated implicitly as (all rhythmic levels imperfect even at higher levels of the modus and maximodus):

```xml
        <mensur xml:id="mensur-L10F1" sign="C" slash="1" tempus="2" prolatio="2" />
```

But at least for the first example above this is not happening, while the third example seems to do this.

**Comment by lpugin** [comment]
The current implementation does require `@tempus` and `@prolatio` to be set even if you have `@sign` and `@slash` given since these are treated as visual only. 

We could change it to infer it as suggested. However, maybe we want to consider taking into account some discussions we had during the MEC in Paderborn. One idea is to have `.log` attributes to be used when the visual and the logical values diverge. This would bring back one idea already brought [here](https://github.com/music-encoding/music-encoding/pull/1015). @martha-thomae, @annplaksin and @pe-ro can say more about it.

This was dicussed for `@num` and `@numbase`, but it could potentially also apply here. At the same time, it seems that there is currently a good consensus on the distinction between visual (`@sign` and `@slash`) attributes and (more) logical ones (`@modus.*`, `@tempus` and `@prolatio`), so this might well be only an implementation decision.

@martha-thomae do you think we should change Verovio behaviour and infer the values? We would still have the option to give it explicitly if the desired value is diverging from the inferred one.

**Reply by martha-thomae** [reply]
I don't see a problem with inferring the "logical" domain attributes from the sign for this particular sign (Cut-C), but maybe it won't be a good idea for others(?) Let me finish reading the discussion and comment on that.

**Comment by craigsapp** [comment]
Here are the defaults that should be used for the standard mensuration signs:

<img width="1030" alt="Screenshot 2023-09-27 at 07 53 42" src="https://github.com/rism-digital/verovio/assets/3487289/1378c23f-c2c4-4358-baf8-cbdfb7c4a39a">
 
Taken from _Plain and easy introduction to practical music_ by Thomas Morley:
https://digital.library.unt.edu/ark:/67531/metadc86
page 17:
https://digital.library.unt.edu/ark:/67531/metadc86/m1/40

The first four mensuration signs (at the bottom of the table) explicitly describe the `@tempus` and `@modusminor` perfections/imperfection, while the last four mensuration signs explicitly decribes the `@prolatio` and `@tempus` values.   The other division levels are implicitly given by the signs, and may not always be the same in actual music, mostly modus maior.   So yes, the defaults should be able to be overridden with explicit `@prolatio`, `@tempus`, `@modusminor` and `@modusmaior`.

Here is the same chart highlighting the explicit and implicit values for the signs:

<img width="751" alt="Screenshot 2023-09-27 at 08 21 41" src="https://github.com/rism-digital/verovio/assets/3487289/3bd4105a-8e7c-4abc-89fd-9432fed1115b">

For the first four mensuration signs, the `O`/`C` indicate the divisions at the modus minor level, while the `2`/`3` numbers indicate the divisions at the tempus level.   For the last four mensuration signs, the `O`/`C` are for tempus divisions, and the dot/lack of dot are for divisions at the prolatio level.   The implicit divisions are all `2` except for the modus maior of `O3` which is perfect.


The first four mensuration signs can have alternate meaning from the standard mensuration signs.   In particular here is the standard meaning of `C3`:

<img width="996" alt="Screenshot 2023-09-27 at 08 05 24" src="https://github.com/rism-digital/verovio/assets/3487289/f86761a9-4ed0-4356-8999-40e3e8a5f492">

`C3` can also mean `C` mensuration with sesquilatera (typically at the tempus level, but also possible on the prolatio level instead):

<img width="995" alt="Screenshot 2023-09-27 at 08 08 04" src="https://github.com/rism-digital/verovio/assets/3487289/be705cdc-7d13-41ba-9842-36e1e370e152">

This is also indicated most commonly with the mensuration sign "3" where the main part of the sign (C or Cut-C) is implied from the prevailing mensuration.  In this case the sesquialtera causes the tempus level to become perfected, but there is also a 3:2 proportion added to the durations on that level, such that 3 semibreves (whole notes) in sesquialtera take the same time as 2 non-sesquialtera semibreves.

Minor sesquialtera is also possible (also indicated by `C3` or `3` but is indistinguishable from major sesquialtera by looking at the sign, and musicians have to look at the following music to see which one fits the music -- i.e, a lot of semiminims (quarter notes) imply that minor sesquialtera is being used):

<img width="995" alt="Screenshot 2023-09-27 at 08 12 36" src="https://github.com/rism-digital/verovio/assets/3487289/a84b91b0-d8ca-4871-a362-69823e797eb1">

-----------------------------


It would be useful for @annplaksin to give the sesquialtera encodings of `<mensur>` for the above two examples using the new `mensur@level` attribute.

Also I am wondering how Cut-C sesquialtera will be encoded (Cut-C has an implied proportion of 2:1 due to the slash for all rhythmic levels, and 3:2 proportion for only the tempor or prolatio rhythmic level):

<img width="980" alt="Screenshot 2023-09-27 at 08 33 45" src="https://github.com/rism-digital/verovio/assets/3487289/314bb1e8-007a-4b89-8453-a6d20acf6b39">

**Reply by lpugin** [reply]
That looks good and seems to be an excellent basis for the documentation. What would be useful to have is some test cases for the eight signs. They could be added to https://github.com/rism-digital/verovio.org/tree/gh-pages/_tests/mensural (or `/mensur`).

Once we have the examples from @annplaksin I can look at the implementation of these.

**Reply by martha-thomae** [reply]
Yes, I don't see anything wrong with this idea. We can encode the sign and then let Verovio infer these mensuration values (for `modusmaior`, `modusminor`, `tempus`, and `prolatio`), and we can add these attributes explicitly if **(A)** we want to or **(B)** they differ from the default. 

We would only need to inform the other members of the IG as that is not the way things have been working so far, but I think it is a good solution. What do you think, @annplaksin? Things wouldn't change that much and we can add this to the documentation once we decide this would be the way to go for Verovio.

**Comment by annplaksin** [comment]
Sorry for arriving late to the party, and being confused as well...
Could we please collect a list of cases that should be covered by examples? 
That would help a lot.

**Reply by lpugin** [reply]
I added eight examples corresponding to the cases given by Morley https://www.verovio.org/test-suite.xhtml?cat=mensur

All mensur signs are coded explicitly, with `@modusmaior`, `modusminor`, `@tempus` and `prolatio`. The `@sign` is treated as visual only. So are the numbers, except for their global impact on the timemap / MIDI. This should be fixed. We can ignore `@num` and `@numbase`, and all these examples would still align properly. The question is more how they should align relatively to each other - or how the tempo should change when passing from one to another.
